### PR TITLE
Add support for old urls

### PR DESCRIPTION
**Pull request by odinuge**

### CODE REVIEW

File: pages/jobbannonse.js [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
 // For client redirects
-const Redirect = props => {
+const Redirect = () => {
   useEffect(() => {
-    Router.replace(props.newLocation);
-  }, [props.newLocation]);
+    Router.replace('/jobb');
+  }, []);
   return null;
 };
```
Comment (LudvigHz) [inline pages/jobbannonse.js]: So no redirect on the client now? (not that it's necessary)

Comment (odinuge) [reply pages/jobbannonse.js]: Are there still any old links that use shallow routing on itdagene.no? If they are present in articles etc. they will not be not use shallow routing anyway, so it shouldn't matter.

File: lib/withData.js [hunk 2]
```diff
@@ -58,8 +58,10 @@ const getOptions = (options: DataOptions, router: NextRouter) => {
 
   return { variables, query };
 };
-const withData = <T>(
-  ComposedComponent: React.ComponentType<WithDataProps<T>>,
+export const withData = <T>(
+  ComposedComponent: React.ComponentType<WithDataProps<T>> & {
+    getInitialProps?: Object => Promise<Object> | Object
```
Comment (odinuge) [inline lib/withData.js]: TODO add typings to this. Didn't care fight more than necessary with flow if some ts is in works.

**Comment by odinuge** [comment]
Wanna merge this, or ill close it in a day or two. Getting annoyed by seeing it every day in the PR list...

**Pull request merged**